### PR TITLE
Added the ability to Delete a recommendation

### LINKED ISSUE #18: Need ability to Delete a recommendation
**As a** manager  
 **I need** to create a service to delete resource information in Database
 **So that** I can maintain a up-to-date database of resources 
   
 ### Details and Assumptions
 * if a resource is not available we need to keep up to date of resource DB so the file should be updated
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given to delete a particular resource from DB
 When a API call is made to delete
 Then the file Database is updated
 ```

### PR DESCRIPTION
**Pull request by freshao1017**
Need ability to Delete a recommendation

### CODE REVIEW

**Comment by aianurag09** [comment]
Changing merge branch from changtk to main
so that we can clearly observe the changed @freshao1017

**Comment by freshao1017** [comment]
> 

I have changed it.

**Comment by freshao1017** [comment]
> > 
> 
> I have changed it.

File: service/routes.py [hunk 1]
```diff
@@ -74,7 +74,19 @@ def create_Recommendation():
 ######################################################################
 # DELETE A RECOMMENDATION
 ######################################################################
+@app.route("/recommendations/<int:item_id>", methods=["DELETE"])
+def delete_recommendation(item_id):
```
Comment (changtk07) [inline service/routes.py]: Should use plural here, `delete_recommendations`

File: tests/test_models.py [hunk 2]
```diff
@@ -112,3 +112,12 @@ def test_add_a_recommendation(self):
...
         recommendations = Recommendation.all()
         self.assertEqual(len(recommendations), 1)
+        
+    def test_delete_a_recommendation(self):
+        """Delete a item"""
+        recommendation = Recommendation(id=1, src_product_id=100, rec_product_id=200, type= "UP_SELL")
+        recommendation.create()
+        self.assertEqual(len(recommendation.all()), 1)
+        # delete the  recommendation and make sure it isn't in the database
+        product.delete()
```
Comment (changtk07) [inline tests/test_models.py]: should be `recommendation`

**Comment by aianurag09** [comment]
You need to have the base branch as changtk, so that we can easily track what updates are made. Hence I changed to changtk, let the base branch be changtk. I'm updating it now @freshao1017

**Review by changtk07 — CHANGES_REQUESTED** [review]
Reviews

**Comment by aianurag09** [comment]
Please make sure you run nosetests to see how the code coverage is

**Comment by freshao1017** [comment]
@aianurag09 Do you want me to merge into main or changtk? You changed my base to changtk

**Review by changtk07 — APPROVED** [review]
Approved

**Review by aianurag09 — APPROVED** [review]

File: tests/test_routes.py [hunk 3]
```diff
@@ -113,3 +113,17 @@ def test_create_recommendation_no_content_type(self):
         """Create a recommendation with no content type"""
         resp = self.app.post(BASE_URL)
         self.assertEqual(resp.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
+        
+    def test_delete_recommendation(self):
+        """Delete a item"""
+        test_recommendation = self._create_recommendation(1)[0]
```
Comment (changtk07) [inline tests/test_routes.py]: `_create_recommendation` -> `_create_recommendations`

**Review by changtk07 — CHANGES_REQUESTED** [review]

**Review by changtk07 — APPROVED** [review]

**Pull request merged**